src/main.rs

```diff
@@ -110,6 +113,46 @@ fn search_snare_conf() -> Option<PathBuf> {
     None
 }
 
+/// If the config specified a 'user' then switch to that and update $HOME and $USER appropriately.
+fn change_user(conf: &Config) {
+    match conf.user {
+        Some(ref user) => match get_user_by_name(&user) {
```

Comment:
It hadn't occurred to me before, but looking up users offers racy scenarios.

For example here, you look up a user by name, getting back a `User` struct containing numeric IDs etc, but that struct could go stale at any moment.

I don't there's a way to fix this though(?)

Reply:
It's an inevitable issue -- there is no global lock on the OS and user! [Imagine if there was -- your computer would have to ban you from working at some points.]

Reply:
Yes. I also concluded that you'd need some kernel change, but a lock would be madness. Perhaps some kind of compare and swap scenario, but now we are off on a tangent.

Reply:
It's absolutely impossible -- you'd have to stop the user doing anything! It would be the lock to end all locks ;)